### ISSUE TITLE
Jobs ending with status "CompletedWithErrors" listed as "Completed"

### ISSUE THREAD

**Issue by markstensgaard** [Azure/azure-storage-azcopy#843]
### Which version of the AzCopy was used? 
10.3.3

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows
### What command did you run?
azcopy cp "\\\CXN7521\Projects\res" "https://redactedstorageaccount.blob.core.windows.net/02239?sv=2019-02-REDACTED-32DNL9RUQpGdQrv9TziFpxc%3D" --recursive=true

### What problem was encountered?
Jobs sometimes fail with final job status from the azcopy command being "CompletedWithErrors". The azcopy return code when this happens is correctly "1". 

However, the "CompletedWithErrors" status does not seem to exist when the job using **azcopy jobs list** (and also **azcopy jobs clean**, which is where I encountered this issue). It is shown as "Completed", when it would seem to be more appropriate to give it "Failed" status or retain its own unique "CompletedWithErrors" status. The help section of **azcopy jobs clean** only lists 'Cancelled', 'Completed', 'Failed', and 'InProgress' as valid status strings.

Example:

**_azcopy cp "\\\CXN7521\Projects\res" "https://redactedstorageaccount.blob.core.windows.net/02239?sv=2019-02-REDACTED-32DNL9RUQpGdQrv9TziFpxc%3D" --recursive=true_**

Total size: 3.95 GB
Start time: 23-Jan-2020 02:49:00

INFO: Scanning...
Job 47edd9c0-af74-7143-7a2d-606886a485ec has started
Log file is located at: C:\Users\markjg\.azcopy\47edd9c0-af74-7143-7a2d-606886a485ec.log

INFO: Trying 4 concurrent connections (initial starting point)
INFO: Trying 16 concurrent connections (seeking optimum)
INFO: Trying 4 concurrent connections (backing off)
INFO: Trying 8 concurrent connections (seeking optimum)
INFO: Trying 16 concurrent connections (seeking optimum)
INFO: Trying 8 concurrent connections (backing off)
INFO: Trying 9 concurrent connections (seeking optimum)
INFO: Trying 8 concurrent connections (at optimum)
INFO:
INFO: Automatic concurrency tuning completed.
INFO:
99.8 %, 6324 Done, 0 Failed, 1 Pending, 0 Skipped, 6325 Total,  (Disk may be limiting speed)

Job 47edd9c0-af74-7143-7a2d-606886a485ec summary
Elapsed Time (Minutes): 43.7486
Total Number Of Transfers: 6325
Number of Transfers Completed: 6324
Number of Transfers Failed: 1
Number of Transfers Skipped: 0
TotalBytesTransferred: 4222370637
Final Job Status: **CompletedWithErrors**

azcopy_return_signal =

     1

**From output of  **_azcopy jobs list_****
JobId: 47edd9c0-af74-7143-7a2d-606886a485ec
Start Time: Thursday, 23-Jan-20 02:49:06 CET
Status: **Completed**
Command: cp \\\CXN7521\Projects\res https://redactedstorageaccount.blob.core.windows.net/02239?sv=2019-02-REDACTED-32DNL9RUQpGdQrv9TziFpxc%3D --recursive=true

### How can we reproduce the problem in the simplest way?
Somehow purposefully run an azcopy job that ends with final job status "CompletedWithErrors", and then check the job status with the **azcopy jobs list** command.

### Have you found a mitigation/solution?
For now, just not cleaning jobs with "Completed" status (also waiting on resolution of [this issue](https://github.com/Azure/azure-storage-azcopy/issues/814)). The script I have written only continues when getting a successful "0" return code from azcopy, so I am able to identify the failed jobs based on the lack of some additional things that would have been put into the containers subsequent to the initial azcopy job, but it would be much better if I could identify the failed jobs through azcopy itself.

**Comment by adreed-msft** [comment]
`azcopy jobs list` seems to just take a high-level look at all of the job headers, rather than to traverse through the individual transfers under the jobs.

This means that under `azcopy jobs list`, you'll probably only ever see these statuses:
`InProgress`
`Canceled`
`Paused`
`Completed`

With more detail coming out of `azcopy jobs status`:

`CompletedWithErrors`
`CompletedWithSkipped`
`CompletedWithErrorsAndSkipped`
`Failed`

**Comment by markstensgaard** [comment]
I do not see "azcopy jobs status" as a possible command in version 10.3.4, and an attempt to run it simply brings up the "azcopy jobs --help" help text.

**_C:\\>azcopy jobs status_**
```
Sub-commands related to managing jobs.

Usage:
  azcopy jobs [command]

Examples:
azcopy jobs show [jobID]

Available Commands:
  clean       Remove all log and plan files for all jobs
  list        Displays information on all jobs
  remove      Remove all files associated with the given job ID.
  resume      Resume the existing job with the given job ID.
  show        Show detailed information for the given job ID

Flags:
  -h, --help   help for jobs

Flags Applying to All Commands:
      --cap-mbps uint32      Caps the transfer rate, in megabits per second. Moment-by-moment throughput might vary slightly from the cap. If this option is set to zero, or it is omitted, the throughput isn't capped.
      --output-type string   Format of the command's output. The choices include: text, json. The default value is 'text'. (default "text")

Use "azcopy jobs [command] --help" for more information about a command.

```

**Comment by adreed-msft** [comment]
Ah, I mistook, I meant `show`

**Comment by markstensgaard** [comment]
I see. I wasn't aware of the high-level status versus the detailed status. I think it would be beneficial to include both with the "list" command. 

I will note that the help text for "azcopy jobs clean" implies that "Failed" should also be a high-level status:

`      --with-status string   only remove the jobs with this status, available values: Cancelled, Completed, Failed, InProgress, All (default "All")`

But, in general, I find it somewhat unexpected behavior that jobs with unsuccessful transfers that got return signals of '1' would get "Completed" status rather than "Failed" status.

Perhaps the simplest solution is to simply upgrade "Failed" from a detailed status to high-level status. Or maybe another high-status label could be added for jobs that are "Completed" but had problems? Or just leave them as "InProgress" until all files are successfully transferred? 

It might also help if the "clean" command could accept the detailed status strings as "--with-status" inputs (and maybe also add an "--except-status" option to the clean command).

**Comment by JohnRusk** [comment]
Might be worked on in 10.5

**Comment by nakulkar-msft** [comment]
Fixed with #1058
